runtime/oti/VMHelpers.hpp

```diff
@@ -768,6 +768,36 @@ class VM_VMHelpers
 	{
 		j9object_t instance = NULL;
 
+		instance = inlineAllocateIndexableObject(currentThread, objectAllocate, arrayClass, size, initializeSlots, memoryBarrier, sizeCheck);
+
+		if (NULL == instance) {
+			instance = currentThread->javaVM->memoryManagerFunctions->J9AllocateIndexableObject(currentThread, arrayClass, size, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+			if (J9_UNEXPECTED(NULL == instance)) {
+				setHeapOutOfMemoryError(currentThread);
+			}
+		}
+		return instance;
+	}
+
+	/**
+	 *
+	 * Perform a non-instrumentable allocation of an indexable flattened or unflattened class.
+	 *
```

Comment:
Add a comment saying "unflattened class that contain the J9ClassContainsUnflattenedFlattenables flag will return NULL"